### PR TITLE
[WIP?] Add mni_ref parameter that allows external reference image for resampling

### PR DESCRIPTION
**Pull request by feilong**
An attempt to solve the issue:
https://github.com/poldracklab/fmriprep/issues/337

### CODE REVIEW

**Comment by effigies** [comment]
Thanks for this. I leave it to @chrisfilo as to whether the exact method for specifying this within the workflow is appropriate, but could you also update the `fmriprep-docker` script, so this can be easily used inside docker?

Options for pulling in files or directories from outside are defined in [`g_wrap`](https://github.com/poldracklab/fmriprep/blob/34fe8d2b0dda060081fc444359cb0df62c27ab0a/wrapper/fmriprep_docker.py#L220-L224), and are then [mapped](https://github.com/poldracklab/fmriprep/blob/34fe8d2b0dda060081fc444359cb0df62c27ab0a/wrapper/fmriprep_docker.py#L313-L315).

Something like:

```Python
g_wrap.add_argument('--mni-ref', required=False, action='store', type=os.path.abspath)

...

if opts.mni_ref:
    target = '/imports/' + os.path.basename(opts.mni_ref)
    command.extend(['-v', ':'.join((opts.mni_ref, target, 'ro'))])
    unknown_args.extend(['--mni-ref', target])
```

If this turns out to be too much of a pain, don't worry, and I'll figure it out once this is merged.

**Comment by feilong** [comment]
Thanks @effigies, I updated (copy/paste) the file as you said.  I'm not familiar with Docker, so I'll leave the rest to the experts.

**Comment by chrisfilo** [comment]
As we discussed in https://github.com/poldracklab/fmriprep/issues/337#issuecomment-300376362 it would make sense to rename this to `--output-grid-reference`. Could you also rebase/merge master?

**Comment by feilong** [comment]
Sure, will do tomorrow.

**Comment by feilong** [comment]
I rebased my commits and renamed the parameter. I tested it locally and the outputs look fine.

File: fmriprep/cli/run.py [hunk 1]
```diff
@@ -92,6 +92,11 @@ def get_parser():
         choices=['MNI152NLin2009cAsym'], default='MNI152NLin2009cAsym',
         help='volume template space (default: MNI152NLin2009cAsym)')
 
+    g_conf.add_argument(
+        '--output-grid-reference', required=False, action='store', default=None,
+        help='Grid reference image for resampling BOLD files to volume template space. '
+             'It determines the grid (i.e., affine matrix) of the output files if provided.')
```
Comment (chrisfilo) [inline fmriprep/cli/run.py]: I would mention that this "determines the field of view and resolution of the output images, but is not used in normalization".

**Pull request merged**